core-js/src/main/javascript/cdf/components/TextInputComponent.js

```diff
@@ -24,9 +24,20 @@ define([
       myself._addHtmlToPlaceholder();
 
       var el = $("#" + myself.name);
-      el.val(myself.dashboard.getParameterValue(myself.parameter));
+      if (myself.formatter) {
```

Comment:
Please read the parameter value a single time and declare the `val` variable. For example:
```js
var value = myself.dashboard.getParameterValue(myself.parameter);
if (myself.formatter) {
  value = myself.formatter.format(value);
}
el.val(value);
```
```

Reply:
Done